huaweicloud/services/acceptance/er/data_source_huaweicloud_er_flow_logs_test.go

```diff
@@ -0,0 +1,307 @@
+
+		byName   = "data.huaweicloud_er_flow_logs.filter_by_name"
+		dcByName = acceptance.InitDataSourceCheck(byName)
+
+		byStatus   = "data.huaweicloud_er_flow_logs.filter_by_status"
+		dcByStatus = acceptance.InitDataSourceCheck(byStatus)
+
+		byEnabled   = "data.huaweicloud_er_flow_logs.filter_by_enabled"
+		dcByEnabled = acceptance.InitDataSourceCheck(byEnabled)
+
+		byLogGroupId   = "data.huaweicloud_er_flow_logs.filter_by_log_group_id"
+		dcByLogGroupId = acceptance.InitDataSourceCheck(byLogGroupId)
+
+		byLogStreamId   = "data.huaweicloud_er_flow_logs.filter_by_log_stream_id"
+		dcByLogStreamId = acceptance.InitDataSourceCheck(byLogStreamId)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceFlowLogs_basic(name, bgpAsNum),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					dcByResourceType.CheckResourceExists(),
+					resource.TestCheckOutput("resource_type_filter_is_useful", "true"),
+
+					dcByResourceId.CheckResourceExists(),
+					resource.TestCheckOutput("resource_id_filter_is_useful", "true"),
+
+					dcByFlowLogId.CheckResourceExists(),
+					resource.TestCheckOutput("flow_log_id_filter_is_useful", "true"),
+
+					dcByName.CheckResourceExists(),
+					resource.TestCheckOutput("name_filter_is_useful", "true"),
+
+					dcByStatus.CheckResourceExists(),
+					resource.TestCheckOutput("status_filter_is_useful", "true"),
+
+					dcByEnabled.CheckResourceExists(),
+					resource.TestCheckOutput("enabled_filter_is_useful", "true"),
+
+					dcByLogGroupId.CheckResourceExists(),
+					resource.TestCheckOutput("log_group_id_filter_is_useful", "true"),
+
+					dcByLogStreamId.CheckResourceExists(),
+					resource.TestCheckOutput("log_stream_id_filter_is_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccFlowLogsDataSource_base_1(name string, bgpAsNum int) string {
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_er_availability_zones" "test" {}
+
+resource "huaweicloud_er_instance" "test" {
+  availability_zones = slice(data.huaweicloud_er_availability_zones.test.names, 0, 1)
+  name               = "%[2]s"
+  asn                = %[3]d
+}
+
+resource "huaweicloud_lts_group" "test" {
+  group_name  = "%[2]s"
+  ttl_in_days = 7
+}
+
+resource "huaweicloud_lts_stream" "test" {
+  group_id    = huaweicloud_lts_group.test.id
+  stream_name = "%[2]s"
+}
+
+resource "huaweicloud_er_vpc_attachment" "test" {
+  instance_id            = huaweicloud_er_instance.test.id
+  vpc_id                 = huaweicloud_vpc.test.id
+  subnet_id              = huaweicloud_vpc_subnet.test.id
+  name                   = "%[2]s"
+  auto_create_vpc_routes = true
+
+  tags = {
+    foo = "bar"
+  }
+}
+`, common.TestVpc(name), name, bgpAsNum)
+}
+
+func testAccFlowLogsDataSource_base_2(name string, bgpAsNum int) string {
```

Comment:
you can merge `testAccFlowLogsDataSource_base_1` and `testAccFlowLogsDataSource_base_2` into `testAccFlowLogsDataSource_base`

Reply:
done